lib/rpa_paycode.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# -*- mode: python3 -*-
+# This file (c) 2020 Jonald Fyookball
+# Part of the Electron Cash SPV Wallet
+# License: MIT
+'''
+This implements the functionality for RPA (Reusable Payment Address) aka Paycodes
+'''
+
+from . import rpaaddr
+from . import bitcoin
+from . import transaction
+from . import address
+from decimal import Decimal as PyDecimal
+from .bitcoin import COIN, TYPE_ADDRESS, sha256 
+from .address import Address, AddressError, Base58, PublicKey
+from .transaction import Transaction
+
+def satoshis(amount):
+    # satoshi conversion must not be performed by the parser
+    return int(COIN*PyDecimal(amount)) if amount not in ['!', None] else amount
+ 
+def resolver(wallet, x, nocheck):
+    if x is None:
+        return None
+    out = wallet.contacts.resolve(x)
+    if out.get('type') == 'openalias' and nocheck is False and out.get('validated') is False:
+        raise BaseException('cannot verify alias', x)
+    return out['address']
+
+def mktx(wallet, config, outputs, fee=None, change_addr=None, domain=None, nocheck=False,
+              unsigned=False, password=None, locktime=None, op_return=None, op_return_raw=None):
+    if op_return and op_return_raw:
+        raise ValueError('Both op_return and op_return_raw cannot be specified together!')
+     
+    domain = None if domain is None else map(self._resolver, domain)
+    final_outputs = []
+    if op_return:
+        final_outputs.append(OPReturn.output_for_stringdata(op_return))
+    elif op_return_raw:
+        try:
+            op_return_raw = op_return_raw.strip()
+            tmp = bytes.fromhex(op_return_raw).hex()
+            assert tmp == op_return_raw.lower()
+            op_return_raw = tmp
+        except Exception as e:
+            raise ValueError("op_return_raw must be an even number of hex digits") from e
+        final_outputs.append(OPReturn.output_for_rawhex(op_return_raw))
+         
+    for address, amount in outputs:
+        address = resolver(wallet, address, nocheck)
+        amount = satoshis(amount)
+        final_outputs.append((TYPE_ADDRESS, address, amount))
+         
+    coins = wallet.get_spendable_coins(domain, config)
+    tx = wallet.make_unsigned_transaction(coins, final_outputs, config, fee, change_addr)
+    if locktime != None:
+        tx.locktime = locktime
+    if not unsigned:
+        run_hook('sign_tx', wallet, tx) 
+        wallet.sign_transaction(tx, password)
+    return tx
+
+def calculate_paycode_shared_secret(private_key, public_key, outpoint):
+    
+    # private key is expected to be an integer.
+    # public_key is expected to be bytes.
+    # outpoint is expected to be a string.
+    # returns the paycode shared secret as bytes
+        
+    from fastecdsa import keys, curve
```

Comment:
Is this module always available and installed?  I don't have it... If not you cannot assume it exists!

We would need to update everything inside out setup.py, requirements, etc for this!!

Is there any way you can do this stuff using `ecdsa`?  Or the existing dependencies we have?!  This is a big deal!

Reply:
If not then this feature will just "not work" unless user installs this, which I guess is ok for now given this is CLI-only and experimental..